src/main/java/alfred/task/Task.java

```diff
@@ -0,0 +1,119 @@
+package alfred.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Encapsulates all Alfred Tasks.
+ */
+public abstract class Task {
+    // class constants
+    public static final String FORMAT_SPLIT = "`";
+    protected static final String COMPLETION_MARK = "X";
+    protected static final String INCOMPLETE_MARK = " ";
+    private static final DateTimeFormatter dateTimeFormatter =
+            DateTimeFormatter.ofLocalizedDateTime(
+                    FormatStyle.MEDIUM, FormatStyle.MEDIUM);
+
+
+    // instance attributes
+    protected final String description;
+    protected boolean completed;
```

Comment:
Perhaps using a prefix such as is to the variable name such as IsCompleted would make it sound more like booleans!